plugins/modules/cloud/misc/proxmox_disk.py

```diff
@@ -0,0 +1,688 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022, Castor Sky (@castorsky) <csky57@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_disk
+short_description: Management of a disk of a Qemu(KVM) VM in a Proxmox VE cluster.
+version_added: 5.5.0
+description:
+  - Allows you to perform some supported operations on a disk in Qemu(KVM) Virtual Machines in a Proxmox VE cluster.
+author: "Castor Sky (@castorsky) <csky57@gmail.com>"
+options:
+  name:
+    description:
+      - The unique name of the VM.
+      - You can specify either I(name) or I(vmid) or both of them.
+    type: str
+  vmid:
+    description:
+      - The unique ID of the VM.
+      - You can specify either I(vmid) or I(name) or both of them.
+    type: int
+  disk:
+    description:
+      - The disk key (C(unused[n]), C(ide[n]), C(sata[n]), C(scsi[n]) or C(virtio[n])) you want to operate on.
+    type: str
+    required: True
+  state:
+    description:
+      - Indicates desired state of the disk.
+      - >
+        I(state=present) intended only for creating new disk (with or without replacing existing disk,
+        see I(force_replace) option).
+      - For changing options in existing disk use I(state=updated)
+      - When I(state=detached) and disk is C(unused[n]) it will be left in same state (not removed).
+    type: str
+    choices: ['present', 'updated', 'grown', 'detached', 'moved', 'absent']
```

Comment:
The value name `gronw` and the logic of the module indicate only the possibility of enlarging the disk. So, just double checking here: is it **not** possible to shrink it?

Reply:
Correct. At the time QEMU have no ability to shrink image.

Reply:
If shrinking is allowed, maybe `size` should be able to handle numbers starting with `-` as well.

Reply:
oops, did not see the response in time :-)

Reply:
> QEMU have no ability to shrink image.

Not quite QEMU, but Proxmox. `qemu-img` can shrink images, but this process needs very carefull preparations.
Proxmox does not support shrinking at all. It has description of `resize` method as `Extend volume size.`

Reply:
How about still naming this `resized` instead of `grown`? Mainly because in case Proxmox ever suddenly start supporting shrinking as well, we don't have to add another state (or create an alias), but the module could just support it out of the box.